spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SqlSpannerQuery.java

```diff
@@ -218,12 +218,30 @@ else if (Pageable.class.isAssignableFrom(paramClass)) {
 
 		resolveSpELTags(queryTagValue);
 
-		return this.spannerOperations.query(this.entityType,
+		return this.isCountOrExistsQuery()
+				? this.spannerOperations.query(struct -> struct.getLong(0),
+						this.entityType, resolveEntityClassNames(queryTagValue.sql),
+						queryTagValue.tags, queryTagValue.params.toArray(),
+						spannerQueryOptions)
+				: this.spannerOperations.query(this.entityType,
 				resolveEntityClassNames(queryTagValue.sql), queryTagValue.tags,
 				queryTagValue.params.toArray(),
 				spannerQueryOptions);
 	}
 
+	@Override
+	protected boolean isCountQuery() {
+		Class returnedType = this.queryMethod.getReturnedObjectType();
+		return returnedType == long.class || returnedType == int.class
```

Comment:
Just like in the Datastore case, we should not make this guess, but support mapping `ResultSet` to a single POJO as a general rule.
Can we support something like this:
```java
@Query("select plate from cars")
List<String> getPlates();
```
Does Spring Data in general support this kind of query methods?